server/src/server.ts

```diff
@@ -216,7 +219,31 @@ let getOpenedFileContent = (fileUri: string) => {
   return content;
 };
 
-process.on("message", (msg: m.Message) => {
+let onMessage = (func: any) => {
+  process.on("message", (msg: m.Message) => {
+    func(messageSender.send, msg);
+  })
+};
+
+let argv = process.argv.slice(2);
+for (let i = 0; i < argv.length; i++) {
```

Comment:
Why are we looping over the args? Is it because you're catering to having the server opened with extra arguments from the debugging mode (does it do that...? I forgot)

Reply:
Thanks for looking at the PR!

It's not because of debug mode: I wasn't aware of that until now and I'm not yet familiar with whatever debugging facilities are provided by either VS code or LSP.

I think I haven't understood your question correctly.  Are you just saying it would be nicer to use something like `process.argv.includes("--stdio")` instead of a loop?

If not: I think you understand all this already, but I'll state the context explicitly.  Some editors, emacs included, want to talk LSP over standard in / standard out (stdio).  Currently rescript-vscode's server.ts, if I've understood right, is only able to talk over "node IPC", which is only intended for use by node programs (i.e. programs like emacs shouldn't try to use that).

I don't know another way for the server process to know it should use stdio than looking at process.argv?  I believe this is standard in LSP servers.  I think I lifted this loop from vscode-languageserver-node:
https://github.com/microsoft/vscode-languageserver-node/blob/53d122831050088171296dc02b57257f5178f0b6/server/src/node/main.ts#L203

Or are you maybe thinking that vscode-languageclient should be used to tell the server to use stdio, even for editors other than VSCode?  If so: I think 1. vscode-languageclient is indended only for use by VSCode clients, not servers 2. All it does other than arranging to use stdin/stdout on the client end is to pass "--stdio" to the server it's starting -- so it's still up to server.ts to test whether "--stdio" is present in the args and implement using that channel.

But as I say I I've probably misunderstood what you're asking, sorry!

Reply:
Nah I mean why not just check the third argument =)

Reply:
lol

OK, only because: 1. I copied the code from vscode-languageserver-node, 2. expecting commandline arguments only at a particular place in argv always seems a bit confusing / error-prone.

Happy to check argv[2] if you prefer, though `.includes("--stdio")` seems nicer?

Reply:
Sure! Could you also simplify `messageSender` and `onMessage` to 2 function calls initiated by whether we see the flag? A little bit of contortion in the code right now

Reply:
Hope that's what you meant

Reply:
No just 2 calls, but I might have missed something. That's fine; I'll merge and do it later.